components/tikv_kv/src/lib.rs

```diff
@@ -265,10 +267,10 @@ pub trait Engine: Send + Clone + 'static {
     /// Local storage engine.
     fn kv_engine(&self) -> Self::Local;
 
-    fn snapshot_on_kv_engine(&self, start_key: &[u8], end_key: &[u8]) -> Result<Self::Snap>;
-
-    /// Write modifications into internal local engine directly.
-    fn modify_on_kv_engine(&self, modifies: Vec<Modify>) -> Result<()>;
+    /// Do some specialization encoding works on `modifies`s according to the
+    /// different implementation of `Engine` before flushing them to the
+    /// underlying kv engine.
```

Comment:
```suggestion
    /// Change the encoding of `Modify` so that they can be written to local
    /// engine directly. After the method, `modifies` should not be passed to
    /// any write methods of `Engine` trait any more.
```